sdr_repository.go

```diff
@@ -74,24 +87,53 @@ func walkSDRs(ctx context.Context, s Session) (SDRRepository, error) {
 	// duplicate it.
 	for getSDRCmd.Req.RecordID != ipmi.RecordIDLast {
 		if err := ValidateResponse(s.SendCommand(ctx, getSDRCmd)); err != nil {
-			// if we get a 0xca or 0xff, we need to implement reservations and
-			// partial reading - hopefully we'll be alright - yet to see a SDR
-			// >70 bytes long - they're specified as 64 after all.
 			return nil, err
 		}
-
-		packet := gopacket.NewPacket(getSDRCmd.Rsp.Payload, ipmi.LayerTypeSDR,
+		headerPacket := gopacket.NewPacket(getSDRCmd.Rsp.Payload, ipmi.LayerTypeSDR,
 			gopacket.DecodeOptions{
 				Lazy: true,
 				// we can't set NoCopy because we reuse getSDRCmd.Rsp
 			})
-		if packet == nil {
-			return nil, fmt.Errorf("invalid SDR: %v", getSDRCmd)
+		if headerPacket == nil {
+			return nil, fmt.Errorf("invalid SDR for record ID %d: empty packet",
```

Comment:
Is this an empty packet, or just one without a recognisable SDR? I would consider taking out `: empty packet`.